vim25/soap/client.go

```diff
@@ -72,11 +72,29 @@ type Client struct {
 	Version   string // Vim version
 	UserAgent string
 
+	// ValidateRootCAs: if set to true, the soap client will ensure the root CA
+	// certificates are valid in SetRootCAs. If at least one of the certs is
+	// invalid, SetRootCAs will return an error.
+	// Defaults to false; which means that the client will silently ignore any
+	// cert validation errors during SetRootCAs.
+	ValidateRootCAs bool
+
 	cookie string
 }
 
 var schemeMatch = regexp.MustCompile(`^\w+://`)
 
+type ErrInvalidCACertificate struct {
```

Comment:
Let's avoid exporting this type, `errInvalidCACertificate` instead would be fine.

Reply:
Well, when we don't export it, how can people who use the package check, if the error was because of an invalid certificate or because something else (reading the file failed ...)?

Reply:
As it is possible to add multiple certs in one go (`cert1:cert2:cert3`) I think it is especially important for users to have access to `ErrInvalidCACertificate.File` to check which of the certs is invalid.

Reply:
I don't see where calling code would make use of the custom error type though, other than in the tests.  If an invalid CA file is provided, the generic `error` interface type with File name in the `Error` message as it is now should be all the user needs.

Just trying to avoid adding to govmomi's too many exported types unless we are certain it is needed.  We can always choose to export it later if there are use case(s) that would require the custom type.  What do you think?